v2/spiffebundle/bundle.go

```diff
@@ -0,0 +1,43 @@
+package spiffebundle
+
+import (
+	"crypto/x509"
+	"io"
+
+	"gopkg.in/square/go-jose.v2"
+)
+
+// Bundle is a SPIFFE bundle.
+type Bundle struct {
+	jose.JSONWebKeySet
+
+	Sequence    uint64 `json:"spiffe_sequence,omitempty"`
+	RefreshHint int    `json:"spiffe_refresh_hint,omitempty"`
+}
+
+// Load loads a Bundle from a file on disk.
+func Load(path string) (*Bundle, error) {
+	panic("not implemented")
+}
+
+// Read decodes a bundle from a reader.
+func Read(r io.Reader) (*Bundle, error) {
+	panic("not implemented")
+}
+
+// Parse parses a bundle from bytes.
+func Parse(b []byte) (*Bundle, error) {
```

Comment:
Similar to Load, do we need a function to go the other way? Marshal? What would a potential `Write()` function need?

Reply:
I waffled about providing a marshal method since it would probably be something like
 ```
func Marshal(bundle *Bundle) ([]byte, error) {
    return json.Marshal(bundle)
}
```

The reason the `Load()` and `Parse()` methods were provided is because they could do validation that the bundle was well-formed.

Reply:
If we hide jose library usage that backs this, then json.Marshal will no longer be viable and we'll need methods to marshal bundles.